### ISSUE TITLE
`scamp` offline mode doesn't provide origins to amplitude processors

### ISSUE THREAD

**Issue by acarapetis** [SeisComP/main#65]
Prior to 5.5.0, `scamp --ep ...` would provide hypocenters to `AmplitudeProcessor::setEnvironment()`. It seems that when the `--picks` option was introduced, the behaviour changed so that `setEnvironment` is now never passed a hypocenter, even when `--picks` is disabled. 

I think the culprit is setting `pick` here: https://github.com/SeisComP/main/blob/master/apps/processing/scamp/amptool.cpp#L741
This means the `if (pick)` here will always be true: https://github.com/SeisComP/main/blob/master/apps/processing/scamp/amptool.cpp#L849

**Comment by gempa-jabe** [comment]
Thank you, I can confirm the bug and a fix is currently  in testing. If you want you can test the following fix yourself and report back:

```diff
diff --git a/apps/processing/scamp/amptool.cpp b/apps/processing/scamp/amptool.cpp
index 57cc45bed..fad7f4ec9 100644
--- a/apps/processing/scamp/amptool.cpp
+++ b/apps/processing/scamp/amptool.cpp
@@ -846,14 +846,15 @@ void AmpTool::process(Origin *origin, Pick *pickInput) {
                        proc->setReferencingPickID(pickID);
 
                        int res = -1;
-                       if ( pick ) {
+                       if ( origin ) {
+                               res = addProcessor(proc.get(), origin, pick.get(), distance, depth, (double) origin->time().value());
+                       }
+                       else {
                                SC_FMT_DEBUG("Measuring {} amplitude for pick {} independent of origin",
                                             ait->c_str(), pick->publicID().c_str());
                                res = addProcessor(proc.get(), nullptr, pick.get(), None, None, None);
                        }
-                       else {
-                               res = addProcessor(proc.get(), origin, pick.get(), distance, depth, (double) origin->time().value());
-                       }
+
                        if ( res < 0 ) {
                                // RecordStream not available
                                if ( res == -2 ) {
```

**Comment by gempa-jabe** [comment]
Fixed in 5.5.5.
